src/main/java/core/basesyntax/FileReaderImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl {
+    public List<String> read(String fileName) {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                lines.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Consider using a logging framework instead of `printStackTrace()` for better error handling and to avoid printing stack traces directly to the console. This will make it easier to manage and track errors in a production environment.